ql/src/java/org/apache/hadoop/hive/ql/optimizer/stats/annotation/StatsRulesProcFactory.java

```diff
@@ -557,14 +557,17 @@ private long evaluateInExpr(Statistics stats, ExprNodeDesc pred, long currNumRow
       }
       for (int i = 0; i < columnStats.size(); i++) {
         long dvs = columnStats.get(i) == null ? 0 : columnStats.get(i).getCountDistint();
-        long intersectionSize = estimateIntersectionSize(aspCtx.getConf(), columnStats.get(i), values.get(i));
+        if (dvs == 0) {
+          factor *= 0.5;
```

Comment:
Can we use [this property](https://github.com/apache/hive/blob/rel/release-4.0.0/ql/src/java/org/apache/hadoop/hive/ql/optimizer/stats/annotation/StatsRulesProcFactory.java#L575) to tune the factor? It could not directly support your original intention, but it will likely work.
If `multiColum` is false, it would work as you expected. If `multiColumn` is true, 0.1 per column could lead to underestimation.

Reply:
sure you could use that as well - that's also better than burning in`.5`

Reply:
@kgyrtkirk Would you mind accepting the proposal and merging this PR? Thanks.